sql/driver/driver_test.go

```diff
@@ -42,100 +41,40 @@ func cleanup(s *server.TestServer, db *sql.DB) {
 	s.Stop()
 }
 
-type resultSlice [][]*string
-
-func (r resultSlice) String() string {
-	results := make([][]string, len(r))
-
-	for i, subSlice := range r {
-		results[i] = make([]string, len(subSlice))
-		for j, str := range subSlice {
-			if str == nil {
-				results[i][j] = "<NULL>"
-			} else {
-				results[i][j] = *str
-			}
-		}
-	}
-
-	return fmt.Sprintf("%s", results)
-}
-
-func asResultSlice(src [][]string) resultSlice {
-	result := make(resultSlice, len(src))
-	for i, subSlice := range src {
-		result[i] = make([]*string, len(subSlice))
-		for j := range subSlice {
-			if subSlice[j] == "<NULL>" {
-				result[i][j] = nil
-			} else {
-				result[i][j] = &subSlice[j]
-			}
-		}
-	}
-	return result
-}
-
-func readAll(t *testing.T, rows *sql.Rows) resultSlice {
+func readAll(rows *sql.Rows) ([][]string, error) {
 	defer rows.Close()
 
 	cols, err := rows.Columns()
 	if err != nil {
-		t.Fatal(err)
+		return nil, err
 	}
 
-	colStrs := make([]*string, len(cols))
-	for i := range cols {
-		colStrs[i] = &cols[i]
+	vals := make([]interface{}, len(cols))
+	for i := range vals {
+		vals[i] = new(sql.NullString)
```

Comment:
Hmm, I don't think we should be using `new` there either. Those instances probably evolved during a conversion from `new(string)` which does make sense because it is otherwise awkward to heap allocate a string (or other basic types).